lib/attr.gi

```diff
@@ -37,3 +37,11 @@ InstallGlobalFunction(DisableAttributeValueStoring, function( attr )
     Info(InfoAttributes + InfoWarning, 2, "Disabling value storing for ",NAME_FUNC(attr));
     SET_ATTRIBUTE_STORING( attr, false);
 end);
+
+CHECK_REPEATED_ATTRIBUTE_SET := function(obj, name, val)
+    if not IsBound(obj!.(name)) then
+        Info(InfoAttributes + InfoWarning, 3, "Attribute ", name, " of ", obj, " is marked as assigned, but it has no value");
+    elif obj!.(name) <> val then
```

Comment:
Note that checking for equality can potentially be very expensive or even run into an infinite loop. So we should be careful to *only* check this if we are going to print a message anyway!

I.e., this function should probably at the very start check if the info level is right, and if not, return immediately.

Reply:
Yes, this is a serious problem, which I will fix.

Reply:
Of course we could also just check for identity, using `IsIdenticalObj`, but then there might be more false warnings... hm